worker-test/src/test/java/com/hpe/caf/worker/workertest/GetWorkerNameIT.java

```diff
@@ -0,0 +1,134 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.worker.workertest;
+
+import org.json.JSONObject;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Consumer;
+import com.rabbitmq.client.ShutdownSignalException;
+import com.rabbitmq.client.Envelope;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.TimeoutException;
+
+
+public class GetWorkerNameIT extends TestWorkerTestBase {
+    private static final String POISON_ERROR_MESSAGE = "could not process the document.";
+    private static final String WORKER_NAME = "TestWorker";
+
+    @Test
+    public void getWorkerNameInPoisonMessageTest() throws IOException, TimeoutException {
+
+        try(final Connection connection = connectionFactory.newConnection()) {
+
+            final Channel channel = connection.createChannel();
+
+            channel.queueDeclare("testworker-out", true, false, false, Collections.emptyMap());
+
+            MyConsumer poisonConsumer = new MyConsumer();
+            channel.basicConsume("testworker-out", true, poisonConsumer);
+
+            final Map<String, Object> retryLimitHeaders = new HashMap<>();
+            retryLimitHeaders.put("x-caf-worker-retry-limit", 3);
+            retryLimitHeaders.put("x-caf-worker-retry", 3);
+
+            final AMQP.BasicProperties properties = new AMQP.BasicProperties.Builder()
+                    .headers(retryLimitHeaders)
+                    .contentType("application/json")
+                    .deliveryMode(2)
+                    .priority(1)
+                    .build();
+
+            final String body = "{\"version\":1,\"taskId\":\"1.1\",\"taskClassifier\":\"testWorkerIdentifier\",\"taskApiVersion\":1," +
+                    "\"taskData\":\"cG9pc29uIG1lc3NhZ2U\",\"taskStatus\":\"RESULT_SUCCESS\",\"context\":{},\"to\":\"worker-in\"," +
+                    "\"tracking\":null,\"sourceInfo\":{\"name\":\"worker-test\",\"version\":\"1.0.0\"}}";
+
+            channel.basicPublish("", "worker-in", properties, body.getBytes(StandardCharsets.UTF_8));
+
+            try {
+                for (int i=0; i<100; i++){
+
+                    Thread.sleep(100);
+
+                    if (poisonConsumer.getLastDeliveredBody() != null){
+                        break;
+                    }
+                }
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+            String returnedBody = new String(poisonConsumer.getLastDeliveredBody(), StandardCharsets.UTF_8);
```

Comment:
What if the loop exits due i >= 100, poisonConsumer.getLastDeliveredBody() will be null